### ISSUE TITLE
Custom attribute scoped CSS

### ISSUE THREAD

**Issue by pndewit** [aurelia/templating#581]
**I'm submitting a feature request**

I am not really sure if this is the right project submit this request in, but I didn't know which one to put it in.
I am building an Aurelia adapter for Material Design and this adapter contains both custom elements and custom attributes. For each component I need to include the component's Material Design CSS file and for elements this is easy (using either the <require from="..."> or the viewResources decorator). But I would really like to accomplish the same thing for my custom attributes.

**Comment by Alexander-Taran** [comment]
and importing css in attribute class does not produce the desired result?

**Comment by pndewit** [comment]
Good point, noticed the css import statements here: http://aurelia.io/docs/tutorials/creating-a-contact-manager#building-the-application-shell.
So I tried something similar like: `import 'bootstrap/dist/css/bootstrap.css';`

But it's not working :(

**Comment by Alexander-Taran** [comment]
@pndewit it should work. how are you bundling/loading?

**Comment by pndewit** [comment]
Don't have a css loader, but using the text loader and injecting the styles myself works:

    import mycss from 'text!bootstrap/dist/css/bootstrap.css';
    
    @inject(DOM, Element)
    export class MyComponent {
      constructor(private dom, private element:Element) {}

      attached() {
        let css = this.dom.injectStyles(mycss);
        this.element.appendChild(css);
      }
    }

**Comment by shahabganji** [comment]
This will work when you know the loader for your css is the `text` loader, however when writing a plugin which requires to load some css in a custom attribute or element, and beforehand you don't know which loader will be used, you need adhere to the standard means of importing CSS such as  `import 'mycss.css'`, that will cause errors when using the built-in bundler and having `text` loader, for you have not specified `text!` on your import.

any solutions to handle this? 

@bigopon @Alexander-Taran

**Comment by huochunpeng** [comment]
See aurelia/cli#976

`import "some.css"` is not a standard JS feature, different bundlers might supported differently. If you are developing aurelia plugin:
1. in html `<require from="some.css"></require>`,
2. in js if no view `@noView([PLATFORM.moduleName('some.css')]);`

Or use inline css string inside JS file, then do `DOM.injectStyles` manually. Like https://github.com/aurelia/dialog/blob/ab7e534c94e2be317e56ec3848185932747eaf34/src/dialog-configuration.ts#L48

**Comment by pndewit** [comment]
@huochunpeng Thanks for the extra options, but I can't use these in my case.
1. and 2. can only be used for `customElement`s and I am trying to create a customAttribute with some extra styles.
Inline CSS string inside JS file in combination with `DOM.injectStyles` would work, but my customAttribute depends on a CSS component from a third party that distributes a CSS file.

**Comment by shahabganji** [comment]
@huochunpeng 

> import "some.css" is not a standard JS feature

My bad, I thought it is.

I currently use `DOM.injectStyles`, and in case of the second approach, I used `viewResources` whcih does not work properly. I'll give noView a shot, but I remember it works with custom elements and not attributes as @pndewit  said. and unfortunately we are also depending on a third party library.

**Comment by pndewit** [comment]
> I'll give noView a shot, but I remember it works with custom elements and not attributes as @pndewit said. and unfortunately we are also depending on a third party library.

I just tested it to be sure and it gets ignored. 😞

**Comment by shahabganji** [comment]
I had seen the issue you mentioned above, I just I have one questions is the built-in bundler using requireJS also depends on webpack? I had the problem with that configuration and not webpack. 

> Hope this clarified some bits.

That definitely clarified, thanks :pray:

**Comment by pndewit** [comment]
Thanks @huochunpeng! 👍

> Otherwise, do nothing, then instruct the users of your plugin to manually do <require from="3rd/party.css"></require> in app.html.

This is what I have been doing for the last year or so. Since I am having a monorepo for Aurelia components (custom- elements and attributes) that is based on another monorepo (also maintained by myself) which contains the actual styles for a Material theme, this is generating quite a long list of imports in the base HTML in some cases. Thus my motivation to search for another solution 😄 

_Note: I am also using RequireJS._

**Comment by huochunpeng** [comment]
Cli requirejs does not use webpack at all. The [PR](https://github.com/aurelia/cli/pull/976) for requirejs setup to support import css in js is not merged yet.
